pilot/pkg/proxy/envoy/v1/config.go

```diff
@@ -318,18 +334,56 @@ func buildRDSRoute(mesh *meshconfig.MeshConfig, node model.Node, routeName strin
 	switch node.Type {
 	case model.Ingress:
 		httpConfigs, _ = buildIngressRoutes(mesh, node, nil, discovery, config)
-	case model.Sidecar, model.Router:
+	case model.Sidecar:
 		instances, err := discovery.GetSidecarServiceInstances(node)
 		if err != nil {
 			return nil, err
 		}
+
 		services, err := discovery.Services()
 		if err != nil {
 			return nil, err
 		}
+
 		httpConfigs = buildOutboundHTTPRoutes(mesh, node, instances, services, config)
 		httpConfigs = buildEgressHTTPRoutes(mesh, node, instances, config, httpConfigs)
 		httpConfigs = buildExternalServiceHTTPRoutes(mesh, node, instances, config, httpConfigs)
+	case model.Router:
+		instances, err := discovery.GetSidecarServiceInstances(node)
+		if err != nil {
+			return nil, err
+		}
+
+		services, err := discovery.Services()
+		if err != nil {
+			return nil, err
+		}
+
+		gateways, err := config.List(model.Gateway.Type, model.NamespaceAll)
+		if err != nil {
+			return nil, err
+		}
+
+		targetGateways := map[string]bool{}
+		for _, spec := range gateways {
+			targetGateways[spec.Name] = true
+		}
+
+		finalConfig := &HTTPRouteConfig{}
+		httpConfigs = buildGatewayHTTPRoutes(mesh, node, instances, services, config, targetGateways)
```

Comment:
@rshriram yeah - but then instead of returning all of the virtual hosts for each of the listeners that match the gateway if we didn't combine them we'd only return one (we make a request here with only one of the many fronting ports). So, we believe you would need to return the full set of hosts / routes depending no matter which port you send to `/v1/routes/listenerPort/x/service-node`